src/utils/daos_hdlr.c

```diff
@@ -1915,34 +1915,32 @@ parse_filename_dfs(const char *path, char **_obj_name, char **_cont_name)
 	return rc;
 }
 
-static ssize_t
+static int
 file_write(struct cmd_args_s *ap, struct file_dfs *file_dfs,
-	   const char *file, void *buf, size_t size)
+	   const char *file, void *buf, daos_size_t *size)
 {
-	ssize_t		num_bytes_written = 0;
 	int		rc = 0;
 	daos_size_t	write_size;
 
 	if (file_dfs->type == POSIX) {
-		num_bytes_written = write(file_dfs->fd, buf, size);
+		write_size = write(file_dfs->fd, buf, *size);
+		*size = write_size;
```

Comment:
Minor: would this be simpler?
```suggestion
		*size = write(file_dfs->fd, buf, *size);
```